xbmc/music/MusicDatabase.cpp

```diff
@@ -525,7 +525,43 @@ bool CMusicDatabase::UpdateAlbum(CAlbum& album)
   SetLibraryLastUpdated();
 
   const std::string itemSeparator = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_musicItemSeparator;
+  bool isBoxset = false;
+  bool canBeBoxset = false;
 
+  if (album.bBoxedSet)
+  {
+    isBoxset = IsAlbumBoxset(album.idAlbum);
+    if (!isBoxset)
+    { // not a boxset already so make sure we have enough discs & they all have titles
+      std::string strSQL;
+      strSQL = PrepareSQL("SELECT iDiscTotal FROM album WHERE idAlbum = %i", album.idAlbum);
+      int numDiscs = static_cast<int>(strtol(GetSingleValue(strSQL).c_str(), nullptr, 10));
+      if (numDiscs >= 2)
+      {
+        canBeBoxset = true;
+        int discValue;
+        for (discValue = 1; discValue <= numDiscs; discValue++)
+        {
+          strSQL =
+              PrepareSQL("SELECT DISTINCT strDiscSubtitle FROM song WHERE song.idAlbum = %i AND "
+                         "song.iTrack >> 16 = %i",
+                         album.idAlbum, discValue);
+          std::string currentTitle = GetSingleValue(strSQL);
```

Comment:
Note to myself really...
What happens if disc number and disc titles are inconsistent so get more than 1 result?
How to handle that JSON has tried to set something as a box set, but the associated song data could prevent it so it doesn't get set? Does it matter this fails quietly? Validate in JSON (not sure we can)?  Or add disc titles "disc N" where they are missing or inconsistent?

Reply:
If there are _some_ titles but not all of them (eg, I tested with a 5 disc compilation with named discs as 1 & 5 and the rest blank) it names the three blank ones. if `!currentTitle.empty(); continue` basically.

Reply:
So set "boxset" when no disc titles and nothing happens but JSON user does not know. Note data updates are asynch, the request response "OK" just means that the request was correctly formulated and there was a Kodi to receive it, not that the data was updated. Perhaps it would be better always fill the blanks even if that is all of them?

Reply:
But I was also wondering if the user has messy tag data such that some songs from disc 3 have disc title X and others Y. `GetSingleValue` will return the first of many, and I guess that could be good enough, but really we would prefer to either sort out the mess (all songs on disc have same disc title) or reject the update.

I'm just in 2 minds.... wanting to make/keep data consistent, not having a means to tell user their update was invalid and not likely to fail silently

Reply:
I guess from a user perspective it depends how you write your stuff.  Personally, and always when testing, I request the data back that I think I've just changed and compare it to see if it really did change or not.  How do other db frontends do this, do they just send the change and assume it was correct, or do a call-back on the data to check it ?  Should Kodi be checking that validity or the program using the API ?
I haven't looked yet, but maybe there is a way to return a "request failed" via json but there again, the calling app has to check for that.

Reply:
All good questions. 
Update is async, so no there is nowhere to return request failed because deeper in Kodi didn't like it. Problem is that consumers don't realise the request "OK" is pretty meaningless, but they should as it is part of the API design. Hence we can say the onus in on the consumer to check update happened like you do in testing. 

But that stance becomes less clear when setting X has a more complex validity. Boxed set validity is possibly the most complex validation the music data has conceptually. Also consumers could be relying on schema validation to enforce things (although that is a silly app design approach).  Bah!

I am tempted here to say we do whatever to protect data consistency, so either
a) reject the update when it doesn't match the internal rules
or 
b) correct all the data to make it match the rules
but don't do a mix

And don't get too clever, because well user mess up their data with silly JSON calls via some app or not, it is down to them.

still do as much as the update wants

Reply:
Well currently, it rejects it if it doesn't fit the rules.  Maybe a notification in the log would be a good idea here, so that at least if someone complains about it, there will be an indication of what/why it failed.  

`if (!canBeBoxset && album.bBoxedSet)
      {
        CLog::Log(LOGNOTICE, "%s : Album with id [%i] does not meet the requirements for a boxset.",
                  __FUNCTION__, album.idAlbum);
        album.bBoxedSet = false;
      }`
This matches the 'rules' for forcing a boxset (at least 2 discs, auto-naming of them if required) by setting the tag in the files, but will fail with a notification if there is only one disc.

Kodi has no way of knowing if any update of songs or albums via nfo or json is actually correct, it just has to do what it's told to do but it'll be smart enough not to make a boxset out of one disc.